src/workbookManager.ts

```diff
@@ -34,31 +34,92 @@ export class WorkbookManager {
                   )
                 : await JSZip.loadAsync(templateFile);
 
-        return await this.generateSingleQueryWorkbookFromZip(zip, query);
+        return await this.generateQuery1WorkbookFromZip(zip, query, docProps);
     }
 
-    private async generateSingleQueryWorkbookFromZip(
+    private async generateQuery1WorkbookFromZip(
         zip: JSZip,
-        query: QueryInfo
+        query: QueryInfo,
+        docProps?: DocProps
     ): Promise<Blob> {
+        await this.updatePowerQueryDocument(zip, query.queryMashup);
+        await this.updateSingleQueryRefreshOnOpen(zip, query.refreshOnOpen);
+        await this.updateDocProps(zip, docProps);
+
+        return await zip.generateAsync({
+            type: "blob",
+            mimeType: "application/xlsx",
+        });
+    }
+
+    private async updatePowerQueryDocument(zip: JSZip, queryMashup: string) {
+        const mashupHandler = new MashupHandler();
         const old_base64 = await pqUtils.getBase64(zip);
-        const new_base64 = await this.mashupHandler.ReplaceSingleQuery(
+        const new_base64 = await mashupHandler.ReplaceSingleQuery(
             old_base64,
-            query.queryMashup
+            queryMashup
         );
         await pqUtils.setBase64(zip, new_base64);
+    }
 
-        if (query.refreshOnOpen) {
-            await this.setSingleQueryRefreshOnOpen(zip);
-        }
+    private async updateDocProps(zip: JSZip, docProps: DocProps = {}) {
+        const { doc, properties } = await documentUtils.getDocPropsProperties(
+            zip
+        );
 
-        return await zip.generateAsync({
-            type: "blob",
-            mimeType: "application/xlsx",
+        //set auto updated elements
+        const docPropsAutoUpdatedElementsArr = Object.keys(
+            docPropsAutoUpdatedElements
+        ) as Array<keyof typeof docPropsAutoUpdatedElements>;
+
+        const nowTime = new Date().toISOString();
+
+        docPropsAutoUpdatedElementsArr.forEach((tag) => {
+            if (
+                properties.getElementsByTagName(
+                    docPropsAutoUpdatedElements[tag]
+                ).length !== 1
```

Comment:
WRT docPropsAutoUpdatedElements - when one is missing does Excel fail to open the workbook? if not - I don't think its invalid.
What happens if there are 2 or more of the same?

Questions are mainly to see we align to document validity Office currently has.

Reply:
They already exists in the default template, we just update them here (dates), 
if there are 2 or more of the same, value get's overrided by the later

Reply:
Since we already have some support for custom templates, input workbook props can vary, best validate how Excel behaves today with these kind of cases and align.

Reply:
To be clear - lets verify these fields are mandatory before we add strict validation for them in templates.